### PR TITLE
Add docstring for `torch.sparse_coo_tensor`

### LINKED ISSUE #7995: documentation for sparse tensor creation after 0.4.0 update
From the 0.4.0 migration guide https://pytorch.org/2018/04/22/0_4_0-migration-guide.html

> ## dtypes, devices and NumPy-style creation functions
> In previous versions of PyTorch, we used to specify data type (e.g. float vs double), device type (cpu vs cuda) and layout (dense vs sparse) together as a “tensor type”. For example, torch.cuda.sparse.DoubleTensor was the Tensor type respresenting the double data type, living on CUDA devices, and with COO sparse tensor layout.
> 
> In this release, we introduce torch.dtype, torch.device and torch.layout classes to allow better management of these properties via NumPy-style creation functions.

Although this introduction to the `dtype`, `device` and `layout` features mentions sparse tensors, there are no examples for sparse tensor creation using the 0.4.0 style. All references I could find online still use things like `torch.sparse.FloatTensor` rather than `torch.tensor`, even in the official `torch.sparse` documentation https://pytorch.org/docs/master/sparse.html . Is there a way to create a sparse tensor using `torch.tensor`? For example, the migration guide mentions examples like `torch.tensor([[1], [2], [3]], dtype=torch.half, device=cuda)` for dense tensors, but for sparse tensors it's not obvious how to use it, since it needs indices, values and size rather than just an array of numbers.

### PR DESCRIPTION
**Pull request by sethah**
Partially fixes https://github.com/pytorch/pytorch/issues/7995.

### CODE REVIEW

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -4018,6 +4018,72 @@ def parse_kwargs(desc):
...
+
+    >>> torch.sparse_coo_tensor(i, v, torch.Size([2,4]), dtype=torch.float64,
+                                device=torch.device('cuda:0'))
+    torch.cuda.sparse.DoubleTensor of size (2,4) with indices:
+    tensor([[ 0,  1,  1],
+            [ 2,  0,  2]], device='cuda:0')
+    and values:
+    tensor([ 3.,  4.,  5.], dtype=torch.float64, device='cuda:0')
+
+    >>> torch.sparse_coo_tensor([], [], torch.Size([])) # Create an empty tensor (of size (0,))
```
[the remark this answers is not in the document]
Comment (li-roy) [reply torch/_torch_docs.py]: I'm not making any significant changes and I'll regenerate this comments on this file if I do.

I'm not sure what's going on here, I think the constructor is broken? It seems like it this is creating a sparse tensor whose indices tensor itself is a sparse tensor.

File: torch/_torch_docs.py [hunk 2]
```diff
@@ -4018,6 +4018,72 @@ def parse_kwargs(desc):
...
+add_docstr(torch.sparse_coo_tensor,
+           r"""
+sparse_coo_tensor(indices, values, size=None, dtype=None, device=None, requires_grad=False) -> Tensor
+
+Constructs a sparse_coo_tensor with non-zero elements at the given :attr:`indices` with the given
+:attr:`values`.
+
+Args:
+    indices (array_like): Initial data for the tensor. Can be a list, tuple,
+        NumPy ``ndarray``, scalar, and other types. Will be cast to a :class:`torch.LongTensor`
```
Comment (ezyang) [inline torch/_torch_docs.py]: @SsnL @zou3519 Is this following the convention for how we document these kinds of arguments?

File: torch/_torch_docs.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ezyang) [reply torch/_torch_docs.py]: Yes, there is something deeply wrong with the printing here. Can someone verify this is the case on master? If so, it sounds like a bug.

[the remark this answers is not in the document]
Comment (sethah) [reply torch/_torch_docs.py]: This stems from the fact that a call to `_indices()` on this empty sparse tensor ends up in the function below.

```cpp
Tensor SparseCPUFloatType::_indices(const Tensor & self) const {
    auto self_ = checked_cast_tensor<SparseCPUFloatTensor>(self.pImpl,"self",1, false);
    if (self_->isScalar()) {
      // Empty tensor
      return self_->type().toScalarType(kLong).tensor({0});
    }
    return Tensor((new CPULongTensor(context, THSFloatTensor_newIndices(self_->tensor))),false);
}
```

In this situation the branch condition is true and `self_->type().toScalarType(kLong).tensor({0})` resolves to `Tensor SparseCPULongType::tensor(IntList size)`. So, the indices are an empty tensor of `SparseCPULongType`. 

I believe @zou3519 wrote the code for this in https://github.com/pytorch/pytorch/pull/4058.

File: torch/_torch_docs.py (hunk 2 above)
Comment (sethah) [reply torch/_torch_docs.py]: Just for reference, the first two sentences are taken directly from the doc for `torch.tensor`. The latter part was added to clarify what type is expected and that indices are _coordinates_. Let me know if I need to change anything.

File: torch/_torch_docs.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (zou3519) [reply torch/_torch_docs.py]: Thanks for pointing that out, @sethah. That is most definitely a bug, I'll send a fix

[the remark this answers is not in the document]
Comment (sethah) [reply torch/_torch_docs.py]: Done.

**Comment by sethah** [comment]
Is there a way to kick off the tests again?

**Review by ezyang — APPROVED** [review]
but you need to fix the lint failure (one liner)

**Comment by soumith** [comment]
thank you @sethah !

**Pull request merged**